src/models/room-state.ts

```diff
@@ -800,7 +802,7 @@ export class RoomState extends TypedEventEmitter<EmittedEvents, EventHandlerMap>
      * @param powerLevel - The power level of the member
      * @returns true if the given power level is sufficient
      */
-    public hasSufficientPowerLevelFor(action: "ban" | "kick" | "redact", powerLevel: number): boolean {
+    public hasSufficientPowerLevelFor(action: keyof Omit<IPowerLevelsContent, "events" | "users">, powerLevel: number): boolean {
```

Comment:
Can we use a type like

```
type KeysMatching<T, V> = {[K in keyof T]-?: T[K] extends V ? K : never}[keyof T];
type Action = KeysMatching<IPowerLevelsContent, number>;

Reply:
nice idea! but as they are we always get `never` as `Action` due to the interface having optional keys, i played around a bit and we'd have to adjust it in one of these ways (that i could think of):

1)  `Required`
```
type KeysMatching<T, V> = {[K in keyof T]-?: T[K] extends V ? K : never}[keyof T];
type Action = KeysMatching<Required<IPowerLevelsContent>, number>;
```

2)  `number | undefined`
```
type KeysMatching<T, V> = {[K in keyof T]-?: T[K] extends V ? K : never}[keyof T];
type Action = KeysMatching<IPowerLevelsContent, number | undefined>;
```

3) `V | undefined`
```
type KeysMatching<T, V> = {[K in keyof T]-?: T[K] extends V | undefined ? K : never}[keyof T];
type Action = KeysMatching<IPowerLevelsContent, number>;
```

i'd opt for the 1st one